src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -835,6 +732,142 @@ public static function stripComments(string $source)
+                    flock($lock, LOCK_UN);
+                    fclose($lock);
+                    $this->container->set('kernel', $this);
+
+                    return true;
+                }
+            }
+        } catch (\Throwable $e) {
+        } finally {
+            error_reporting($errorLevel);
+        }
+
+        return false;
+    }
+
+    private function errorHandlerFunction(array &$collectedLogs, \Closure &$previousHandler): \Closure
+    {
+        return function ($type, $message, $file, $line) use (&$collectedLogs, &$previousHandler) {
+            if (E_USER_DEPRECATED !== $type && E_DEPRECATED !== $type) {
+                return $previousHandler ? $previousHandler($type, $message, $file, $line) : false;
+            }
+
+            if (isset($collectedLogs[$message])) {
+                ++$collectedLogs[$message]['count'];
+
+                return null;
+            }
+
+            $backtrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 5);
+            // Clean the trace by removing first frames added by the error handler itself.
+            for ($i = 0; isset($backtrace[$i]); ++$i) {
+                if (isset($backtrace[$i]['file'], $backtrace[$i]['line']) && $backtrace[$i]['line'] === $line && $backtrace[$i]['file'] === $file) {
+                    $backtrace = \array_slice($backtrace, 1 + $i);
+                    break;
+                }
+            }
+            for ($i = 0; isset($backtrace[$i]); ++$i) {
+                if (!isset($backtrace[$i]['file'], $backtrace[$i]['line'], $backtrace[$i]['function'])) {
+                    continue;
+                }
+                if (!isset($backtrace[$i]['class']) && 'trigger_deprecation' === $backtrace[$i]['function']) {
+                    $file = $backtrace[$i]['file'];
+                    $line = $backtrace[$i]['line'];
+                    $backtrace = \array_slice($backtrace, 1 + $i);
+                    break;
+                }
+            }
+
+            // Remove frames added by DebugClassLoader.
+            for ($i = \count($backtrace) - 2; 0 < $i; --$i) {
+                if (\in_array(
+                    $backtrace[$i]['class'] ?? null,
+                    [DebugClassLoader::class, LegacyDebugClassLoader::class],
+                    true
+                )) {
+                    $backtrace = [$backtrace[$i + 1]];
+                    break;
+                }
+            }
+
+            $collectedLogs[$message] = [
+                'type' => $type,
+                'message' => $message,
+                'file' => $file,
+                'line' => $line,
+                'trace' => [$backtrace[0]],
+                'count' => 1,
+            ];
+
+            return null;
+        };
+    }
+
+    private function removeLegacyContainer(?\ReflectionClass $oldContainer): void
+    {
+        if ($oldContainer && \get_class($this->container) !== $oldContainer->name) {
```

Comment:
I think you should wrap the whole function with an if instead of calling it early here

Reply:
done